src/rpdk/core/jsonutils/utils.py

```diff
@@ -103,6 +103,95 @@ def traverse(document, path_parts):
     return document, tuple(path), parent
 
 
+# pylint: disable=R0913,W0102,C0301
+# flake8: noqa
+def traverse_list_safe(
+    document,
+    path_parts,
+    parent=None,
+    path=None,
+    paths=None,
+    documents=None,
+    parents=None,
+):
+    """Traverse the document according to the reference.
```

Comment:
How does this differ from traverse? this comment makes it seem like its the same

Reply:
forgot to update the comments. I will update them in the next iteration. Essentially, this method allows you to traverse through a nested list. Lets say for example you have a resource which has writeOnlyProperties which are list of strings in an object. To traverse the list of writeOnlyProperty list you could provide * to denote all the indices of the list and traverse each of them

Reply:
what is the `safe` distinction here? why do we need a "safe" method, thus presumably our standard method which is "unsafe"?

Reply:
All this means is that this method should be able to iterate all the elements of the list